### PR TITLE
Add optional fields

### LINKED ISSUE #160: Add additional fields by submitting the osc form
The open source code form only works for the mandatory fields. We can also add the optional fields.

### PR DESCRIPTION
**Pull request by j-rewerts**
Closes #160 . 

This is a pretty fiddly PR. I've gone through and pulled all of the optional fields. It doesn't set the optional field if the input element is empty.

Check out PR #239 for it working on the optional languages field.

### CODE REVIEW

**Comment by j-rewerts** [comment]
The failure is because I'm using a custom attribute called `data-for` on the language checkbox and text boxes. It makes the code for reading these values quite nice.

@imadeDev Is there any way to make `html5validator` allow custom attributes on HTML elements that you know of?

**Comment by imadeDev** [comment]
> The failure is because I'm using a custom attribute called `data-for` on the language checkbox and text boxes. It makes the code for reading these values quite nice.
> 
> @imadeDev Is there any way to make `html5validator` allow custom attributes on HTML elements that you know of?

I will check this issue

File: _includes/codeForm.html [hunk 1]
```diff
@@ -121,7 +121,7 @@ <h2>{{ site.LanguagesCodeDesc[page.lang] }}</h2>
     {% assign languages_array = languages_array | uniq  %}
     {% for language in languages_array %}
       <div class="checkbox" >
-        <label><input class="{{ language }}" type="checkbox" value="{{ language }}" name="{{ language }}"> {{ language  }} </label>
+        <label><input class="{{ language }}" datafor="languages" type="checkbox" value="{{ language }}" name="{{ language }}"> {{ language  }} </label>
```
Comment (fredericlestage-vdm) [inline _includes/codeForm.html]: If you want to simplify and not use the --ignore-re ..., you can use the accepted syntax for data attributes, which is data-attributename="value" (you just need to add an hyphen, i.e. "data-for") and it should be all good

**Comment by fredericlestage-vdm** [comment]
I added a comment concerning the html validation, maybe it could simplify things

**Comment by j-rewerts** [comment]
@fredericlestage-vdm Thanks for catching that!

**Pull request merged**